relay-server/src/utils/split_off.rs

```diff
@@ -0,0 +1,39 @@
+use itertools::Either;
+
+/// Splits off items from a vector matching a predicate.
+pub fn split_off<T, S>(data: Vec<T>, mut f: impl FnMut(T) -> Either<T, S>) -> (Vec<T>, Vec<S>) {
+    let mut split = Vec::new();
+
+    let data = data
+        .into_iter()
+        .filter_map(|item| match f(item) {
+            Either::Left(item) => Some(item),
+            Either::Right(p) => {
+                split.push(p);
+                None
+            }
+        })
+        .collect();
+
+    (data, split)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_split_off() {
+        let data = vec!["apple", "apple", "orange"];
+
+        let (apples, oranges) = split_off(data, |item| {
+            if item != "orange" {
+                Either::Left(item)
+            } else {
+                Either::Right(item)
+            }
```

Comment:
Personally I would find this easier to use if the predicate returned `true` or `false`.

Reply:
A boolean doesn't allow for a different type (partition + map)

Reply:
Right. Should we introduce a dedicated enum then? Something like `SplitOff` vs. `Keep`? Not a blocker, just bikeshedding here.

Reply:
Introduced a `split_off` which takes the boolean and delegates to the `split_off_map` now. I did like the `Either` since it matches the return value `(left, right)`.